File: Add Binary.cpp
class Solution {
public:
    string addBinary(string a, string b) {
        reverse(a.begin(), a.end()) ;
        reverse(b.begin(), b.end()) ;

        int a_len = a.length() , b_len = b.length() ;
        if ( a_len > b_len ) {
            int k = a_len - b_len ;
            while (k) {
                b.push_back('0') ;
                k-- ;
            }
        }
        else if ( a_len < b_len ) {
            int k = b_len - a_len ;
            while (k) {
                a.push_back('0') ;
                k-- ;
            }
        }


        string res ;
        int carry = 0 ;
        for ( int i = 0 ; i < a.length() ; i++ ) {
            int p , q , num ;
            p = (a[i] == '0') ? 0 : 1 ;
            q = (b[i] == '0') ? 0 : 1 ;

            num = p + q + carry ;
            if (num == 3) {
                carry = 1 ;
                res.push_back('1') ;
            }
            else if (num == 2) {
                carry = 1 ;
                res.push_back('0') ;
            }
            else if (num == 1) {
                carry = 0 ;
                res.push_back('1') ;
            }
            else if (num == 0) {
                carry = 0 ;
                res.push_back('0') ;
            }
        }

        if (carry > 0) {
            res.push_back('1') ;
        }

        reverse(res.begin(), res.end()) ;

        return res ;

    }
};

File: Add Binary.java
class Solution {
    public String addBinary(String a, String b) {
        String res = "";

        Integer number = 0;

        Integer i = a.length() - 1, j = b.length() - 1;
        while (i >= 0 || j >= 0 || number == 1) {

            number += ((i >= 0) ? a.charAt(i) - '0' : 0);
            number += ((j >= 0) ? b.charAt(j) - '0' : 0);

            res = (char)(number % 2 + '0') + res ;

            number /= 2;

            i--; j--;
        }

        return res ;
    }
}